File: LEDWall/src/Applications/Licht/Licht.cpp
#include <Arduino.h>
#include "Licht.h"



Licht::Licht(){

}

void Licht::onCreate(SystemInterface* systeminterface){
  systeminterface->ledtisch->clear();
  systeminterface->ledtisch->setcolor(255,255,255);
  systeminterface->ledtisch->show();
}
void Licht::onRun(SystemInterface* systeminterface){

}
void Licht::onDataReceive(String data,SystemInterface* systeminterface){

  if(data == "LichtSchalter"){

    if(lightstatus==false){
    systeminterface->ledtisch->setcolor(255,0,0);
    systeminterface->ledtisch->rect(0,6,10,7);

    systeminterface->ledFeld->setcolor(0,255,0);
    systeminterface->ledFeld->rect(1,1,3,3);

    systeminterface->lightring->light(6,255,255,0);
    
    lightstatus=true;
    }else{
      lightstatus=false;
          systeminterface->ledtisch->setcolor(0,0,0);
    systeminterface->ledtisch->rect(0,6,10,7);

        systeminterface->ledFeld->setcolor(0,0,0);
    systeminterface->ledFeld->rect(1,1,3,3);

        systeminterface->lightring->light(6,0,0,0);

    }
    systeminterface->ledtisch->show();
    
    systeminterface->ledFeld->show();

    systeminterface->lightring->show();
  }
}
void Licht::onStop(SystemInterface* systeminterface){

}

 String Licht::getName(){
  return "Licht";
}

File: LEDWall2.0/src/main/java/de/ft/ledwall/nativeapps/licht/Licht.java
package de.ft.ledwall.nativeapps.licht;

import de.ft.ledwall.Application;
import de.ft.ledwall.SystemInterface;
import org.jetbrains.annotations.NotNull;

public class Licht implements Application {
    @Override
    public void onCreate() {
        SystemInterface.table.clear();
    }

    @Override
    public void onDraw() {
        SystemInterface.table.setColor(255,0,0);
        SystemInterface.table.clear();
        SystemInterface.table.drawPixel(SystemInterface.camera.getXposition(),SystemInterface.camera.getYposition());
    }

    @Override
    public void onRun() {

    }

    @Override
    public void onDataReceive(@NotNull String data, int playerID) {

    }

    @NotNull
    @Override
    public String getName() {
        return "Licht";
    }

    @Override
    public void onStop() {

    }

    @Override
    public int getVersion() {
        return 0;
    }

    @NotNull
    @Override
    public String getUUID() {
        return "ad86fc0a-7056-11eb-8638-0242ac110002";
    }
}
